kong/api/routes/certificates.lua

```diff
@@ -98,13 +101,129 @@ return {
       })
     end,
 
-
     PUT = function(self, dao_factory, helpers)
-      return crud.put(self.params, dao_factory.ssl_certificates)
+      -- no id present, behaviour should be same as POST
+      if not self.params.id then
+        return create_certificate(self, dao_factory, helpers)
+      end
+
+      --id present in body
+
+      -- check if exists
+      local ssl_cert, err = dao_factory.ssl_certificates:find {
+        id = self.params.id
+      }
+
+      if err then
+        return helpers.yield_error(err)
+      end
+
+      if not ssl_cert then
+        return helpers.responses.send_HTTP_NOT_FOUND()
+      end
+
+      local snis = nil
```

Comment:
No need for such `nil` assignments when declaring variables - they will be `nil` already and we'll keep the code less cluttered :)